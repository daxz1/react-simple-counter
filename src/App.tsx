import React, { useState } from 'react';
import Button from '@material-ui/core/Button';
import Grid from '@material-ui/core/Grid';
import Typography from "@material-ui/core/Typography/Typography";

export const App:React.FC = () => {
  const [count, setCount] = useState<number>(0);
  const increment = () => setCount(prevState => prevState + 1);
  const decrement = () => setCount(prevState => prevState - 1);
  return (
    <Grid
        container
        spacing={3}
        direction={"column"}
        alignItems={"center"}
        justify={"center"}
    >
        <Grid item xs={12}>
            <Typography variant="h2">Simple React Counter</Typography>
        </Grid>
        <Grid item xs={12}>
            <Typography variant="body2">
                A simple example of using {' '}
                <a href={'https://reactjs.org/docs/hooks-intro.html'} target='_blank'>React.Hooks</a> to store local state.
                Source Code can be found <a href={'https://github.com/daxz1/react-simple-counter'} target="_blank">here</a>.
            </Typography>
        </Grid>
        <Grid item xs={12}>
            <Button variant="contained" color="primary" onClick={increment}>
                <span data-testid='increment'>+</span>
            </Button>
            <Typography align={"center"}>
                <span data-testid='counter'>{count}</span>
            </Typography>
            <Button variant="contained" color="primary" onClick={decrement}>
                <span data-testid='decrement'>-</span>
            </Button>
        </Grid>
    </Grid>
  );
};

