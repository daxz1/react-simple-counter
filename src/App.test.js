import React from 'react';
import {cleanup, fireEvent, render, screen} from '@testing-library/react';
import {App } from "./App";

afterEach(cleanup);

test('Renders App', () => {
  render(<App />);
  const linkElement = screen.getByText(/Simple React Counter/i);
  expect(linkElement).toBeInTheDocument();
});

test('Snapshot', () => {
  // https://testing-library.com/docs/react-testing-library/api/#asfragment
  const { asFragment } = render(<App />);
  expect(asFragment(<App />)).toMatchSnapshot()
})

test('Init Counter to 0', () => {
  const { getByTestId } = render(<App />);
  expect(getByTestId('counter')).toHaveTextContent('0');
});

test('Increment Counter', () => {
  const { getByTestId } = render(<App />);
  fireEvent.click(getByTestId('increment'));
  expect(getByTestId('counter')).toHaveTextContent('1');
})

test('Decrement Counter', () => {
  const { getByTestId } = render(<App />);
  fireEvent.click(getByTestId('decrement'));
  expect(getByTestId('counter')).toHaveTextContent('-1');
})