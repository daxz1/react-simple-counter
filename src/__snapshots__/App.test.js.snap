// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3 MuiGrid-direction-xs-column MuiGrid-align-items-xs-center MuiGrid-justify-xs-center"
  >
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
    >
      <h2
        class="MuiTypography-root MuiTypography-h2"
      >
        Simple React Counter
      </h2>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
    >
      <p
        class="MuiTypography-root MuiTypography-body2"
      >
        A simple example of using  
        <a
          href="https://reactjs.org/docs/hooks-intro.html"
          rel="noreferrer"
          target="_blank"
        >
          React.Hooks
        </a>
         to store local state. Source Code can be found 
        <a
          href="https://github.com/daxz1/react-simple-counter"
          rel="noreferrer"
          target="_blank"
        >
          here
        </a>
        .
      </p>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
    >
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          <span
            data-testid="increment"
          >
            +
          </span>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
      <p
        class="MuiTypography-root MuiTypography-body1 MuiTypography-alignCenter"
      >
        <span
          data-testid="counter"
        >
          0
        </span>
      </p>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          <span
            data-testid="decrement"
          >
            -
          </span>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</DocumentFragment>
`;
